controllers/vrg_volrep.go

```diff
@@ -1899,16 +1890,31 @@ func (v *VRGInstance) checkPVClusterData(pvList []corev1.PersistentVolume) error
 	return nil
 }
 
-func (v *VRGInstance) restoreClusterDataObjects(objList []client.Object, objType string) (int, error) {
+func restoreClusterDataObjects[
+	ObjectType any,
+	ClientObject interface {
+		*ObjectType
+		client.Object
+	},
+](
+	v *VRGInstance,
+	objList []ObjectType, objType string,
+	cleanupForRestore func(*ObjectType),
+	validateExistingObject func(*ObjectType) error,
+) (int, error) {
 	numRestored := 0
 
-	for _, obj := range objList {
-		v.cleanupForRestore(obj)
-		v.addRestoreAnnotation(obj)
+	for i := range objList {
+		object := &objList[i]
+		objectCopy := &*object
```

Comment:
Why is this a copy? I think objectCopy == object == &objList[i]. They all point to the same memory location.